src/components/ViewArticle.jsx

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom"
+import { fetchArticleByID } from "../utils/api";
+import { LoadingMsg } from "../utils/messages"
+
+
+
+
+
+
+function ViewArticle () {
+
+  const [ article, setArticle ] = useState({})
+  const [ isLoading, setIsLoading ] = useState(true)
+
+
+
+  const navigate = useNavigate()
+
+  let { article_id } = useParams()
+
+  useEffect(()=>{
+    fetchArticleByID(article_id).then(({data})=>{
+      setArticle(data.article)
+      setIsLoading(false)
+    })
+  },[article_id])
+
+  if (isLoading) {
```

Comment:
We needs some handling of errors. 
At the moment, if i go to an article id that doesnt exist, i just get a constant loading page. 

Need a little 404 to let us know something has gone wrong, and its no longer loading